src/test/java/space/obminyashka/items_exchange/end2end/UserFlowTest.java

```diff
@@ -217,6 +218,26 @@ void loginNewUserViaOauth2_shouldCreateValidUser() throws Exception {
                 .andExpect(jsonPath("$.lastName").value(USER_LAST_NAME));
     }
 
+    @Test
+    @Commit
+    @DataSet("user/login_with_oauth2_init.yml")
+    @ExpectedDataSet(value = "user/login_with_oauth2.yml", orderBy = {"created", "name"},
+            ignoreCols = {"id", "password", "created", "updated", "last_online_time"})
+    @WithMockUser(username = NEW_USER_EMAIL)
+    void loginNewUserViaOauth2_shouldFindAndSetValidatedEmail() throws Exception {
+        var oauth2User = createDefaultOidcUser();
+        var user = userService.loginUserWithOAuth2(oauth2User);
```

Comment:
It's terrible practice to invoke two methods as the same time. You should either mock it or define needed data via dataset.